### PR TITLE
space wizard implementation

### PR DESCRIPTION
**Pull request by mitchharpur**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Issue #51 specifies several wizards that participate in the creation of a space , project, pipeline etc. This pull request implements the initial experience for that process. It does not implement the space creation yet but places the hooks to do so.

* **What is the current behavior?** (You can also link to an open issue here)
Wizards not implemented

* **What is the new behavior (if this is a feature change)?**
Wizards experience implemented, but no integration with backend yet. Some refactoring and refinement still needed in subsequent pull request. This pull request is focused on the experience.

* **Other information**:

### CODE REVIEW

**Comment by pmuir** [comment]
ok to [test]

File: src/app/home/home.component.html [hunk 1]
```diff
@@ -62,8 +62,8 @@
   </div>
 </div>
 <!-- Create Space modal -->
-<modal #createSpace title="Create a new collaboration space">
-  <modal-content>
-    <space-dialog></space-dialog>
+<modal #createSpace class="chromeless-modal"  >
```
Comment (pmuir) [inline src/app/home/home.component.html]: @mitchharpur something we should work out how to do (doesn't need to block this PR) is to put the modal in to the <space-wizard>

File: src/app/models/wizard.ts [hunk 2]
```diff
@@ -0,0 +1,233 @@
+//TODO: refactor and refine domain model for wizard and configurator
+import { Space } from './space'
+
+export class Wizard {
```
Comment (pmuir) [inline src/app/models/wizard.ts]: I think we are following the approach of one class per file.

File: src/app/space-wizard/space-wizard.component.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+@Component({
+  host: {
+    'class': 'wizard-container'
+  },
+  selector: 'space-wizard',
+  templateUrl: './space-wizard.component.html',
+  styleUrls: ['./space-wizard.component.scss']
+})
+export class SpaceWizardComponent implements OnInit {
+  configurator: SpaceConfigurator
```
Comment (pmuir) [inline src/app/space-wizard/space-wizard.component.ts]: Missing ; here a couple of times.

**Comment by pmuir** [comment]
Replaced with #80

File: src/app/models/wizard.ts (hunk 2 above)
Comment (mitchharpur) [inline src/app/models/wizard.ts]: yes i figured ... i still wanted to rename/refactor/remove some classes and I have some better names for a more refined and clearer abstraction so just put it all in the wizard bucket untill ...

File: src/app/home/home.component.html (hunk 1 above)
Comment (mitchharpur) [inline src/app/home/home.component.html]: agreed ...the modal paradigm we have still needs work i.m.o ...  chromeless dialogs typically defer look/feel/control to the contained component making things very simple ... the current paradigm splits this functionality into the host dialog/component making it complicated to control look and feel... a modal should ideally follow the single responsibility principle of just making the component modal and offer lifecycle hooks

**Pull request closed without merging**